consensus/src/round_manager.rs

```diff
@@ -243,6 +246,10 @@ pub struct RoundManager {
     fast_rand_config: Option<RandConfig>,
     // Stores the order votes from all the rounds above highest_ordered_round
     pending_order_votes: PendingOrderVotes,
+    // Round manager broadcasts fast shares when forming a QC or when receiving a proposal.
+    // To avoid duplicate broadcasts for the same block, we keep track of blocks for
+    // which we recently broadcasted fast shares.
+    blocks_with_broadcasted_fast_shares: LruCache<HashValue, ()>,
```

Comment:
For each block, validators only send order votes / two-chain vote once, and they should send fast share only once, so we don't need this buffer?

Reply:
As we are observing in forge tests, sometimes due to network delays/out of order messages, some validators are getting next proposal before sending order votes for the previous block. So, I'm calling the `broadcast_fast_shares` function both in `process_verified_proposal` and when new QC is aggregated. So, we will be broadcasting fast shares based on whichever event happens first. Using this buffer to make sure we don't broadcast fast shares for a block twice.